app/routes.py

```diff
@@ -1,2 +1,210 @@
+    
+    goal.title = form_data["title"]
+
+    db.session.commit()
+
+    return {
+        "goal": goal.to_json()
+    }, 200
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"], strict_slashes=False)
+def delete_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({
+        "details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'
+    })
+
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False) 
+def tasks_index():
+    sort_type = request.args.get("sort")
+
+    if sort_type == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    elif sort_type == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    else:
+        tasks = Task.query.all()
+    
+    tasks_response = []
+
+    for task in tasks:
+        tasks_response.append(task.to_json())
+
+    return jsonify(tasks_response), 200
+    
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def get_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    return {
+        "task": task.to_json()}, 200
+        
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_task():
+    request_body = request.get_json()
+
+    try:
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+
+    except KeyError:
+        return make_response({
+            "details": "Invalid data"
+        }, 400)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {
+        "task": new_task.to_json()
+    }, 201
+
+@tasks_bp.route("/<task_id>", methods=["PUT"], strict_slashes=False)
+def update_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    form_data = request.get_json()
+
+    task.title = form_data["title"]
+    task.description = form_data["description"]
+    task.is_complete = form_data["completed_at"]
+
+    db.session.commit()
+
+    return {
+        "task": task.to_json()
+    }, 200
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"], strict_slashes=False)
+def delete_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response({
+        "details": f'Task {task.task_id} "{task.title}" successfully deleted'
+    })
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"], strict_slashes=False)
+def task_mark_complete(task_id):
+    task = Task.query.get_or_404(task_id)
+    current_datetime = datetime.utcnow()
+    
+    task.completed_at = current_datetime
+    
+    db.session.commit()
+
+    response = requests.get('https://slack.com/api/chat.postMessage',\
+            params={
+            "channel": "task-notifications",
+            "text": f"Someone just completed the task {task.title}"},
+            headers={'Authorization': os.environ['SLACK_TOKEN']})
```

Comment:
This should have been named `SLACK_API_KEY` (to match our .env in Learn) that's why the tests failed:

```suggestion
            headers={'Authorization': os.environ['SLACK_API_KEY']})
```